x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/store/KibanaOwnedReservedRoleDescriptors.java

```diff
@@ -127,7 +127,9 @@ static RoleDescriptor kibanaSystem(String name) {
                 // APM telemetry queries APM indices in kibana task runner
                 RoleDescriptor.IndicesPrivileges.builder().indices("apm-*").privileges("read", "read_cross_cluster").build(),
                 RoleDescriptor.IndicesPrivileges.builder().indices("logs-apm.*").privileges("read", "read_cross_cluster").build(),
+                RoleDescriptor.IndicesPrivileges.builder().indices("logs-apm-*").privileges("read", "read_cross_cluster").build(),
```

Comment:
Regarding `logs-apm.*` vs `logs-apm-*` and `metrics-apm.*` vs `metrics-apm-*`, do we need both? What is the difference between indices that have a `.` vs a `-`? And would `logs-apm*` and `metrics-apm*` work, or is this too broad?

Reply:
@elastic/apm-server Perhaps you can help out here. It looks like we at least need to provide permissions to `metrics-apm-*` (see https://github.com/elastic/kibana/issues/170031#issuecomment-1800439472).

Reply:
Since @sqren found out that these permissions are indeed required, and, considering that these indexes are configurable by the user, I would recommend to give kibana/system role access to `metrics-apm*`, `traces-apm*`, and `logs-apm*` as suggested earlier by @jeramysoucy.

I can modify the PR if that's something you would be fine with.

Reply:
IIUC, the issue here is related to using non-standard data stream names (https://github.com/elastic/kibana/issues/170031#issuecomment-1794831960). If that's correct, then should we really be making changes to standard roles? I think customising one should probably require customising the other.

Reply:
> IIUC, the issue here is related to using non-standard data stream names (https://github.com/elastic/kibana/issues/170031#issuecomment-1794831960). If that's correct, then should we really be making changes to standard roles? I think customising one should probably require customising the other.

I second this. We like to be conservative with making any changes to the `kibana_system` role, especially concerning indices that are not explicitly internal or owned by kibana, or if there is potential for granted access to overlap with user indices.

Reply:
> What would be a solution to the original problem?

I don't actually know why `kibana_system` needs to read from the indices, as opposed to the authenticated user. I'll have to defer to @elastic/apm-ui to answer.

> APM settings in kibana allows customizing indexes with no indication that it has to follow any specific pattern, nor does documentation here indicates that (https://www.elastic.co/guide/en/kibana/current/apm-spaces.html).

That sounds like a bug in the documentation -- we should expect examples to work out of the box.

Reply:
> I don't actually know why kibana_system needs to read from the indices, as opposed to the authenticated user. I'll have to defer to @elastic/apm-ui to answer.

We use the kibana_system role to collect high level telemetry in a background process, which is why it needs access to APM data.

Reply:
@axw 
see also @sqren's comment on the original issue: https://github.com/elastic/kibana/issues/170031#issuecomment-1800439472

Reply:
Thanks, I see. I still don't think the change aligns with how data streams are expected to be named though.

The data stream name has the structure `{data_stream.type}-{data_stream.dataset}-{data_stream.namespace}`, and changing either of the first two components may break applications. The third component (namespace) can be changed arbitrarily, and exists for partitioning data for RBAC, retention, etc.

So maybe the solution is just to improve the documentation, and to update the APM UI's settings page to restrict which data stream patterns can be specified?

Reply:
@axw I know, that's why when I named our indexes I followed this convention: https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme

E.g. our metric index is called `metrics-apm-somekibnaspace`. But `kibana_system` doesn't have permissions to read it and that's what this PR is trying to solve.

`kibana_system` role has permissions to read `traces-apm-*`, but not `logs-apm-*` and `metrics-apm-*`. It seems reasonable for me to allow that.

Reply:
I agree that once `kibana_system` has expected permissions, UI should indicate that the following pattern is expected for APM indexes: `{type}-apm-*` and documentation mentioned above should be changed.

Reply:
@axw given that proposed PR adds permissions to read data streams of the standard structure that you mentioned, do you think it makes sense to merge it, or are there any other objections I haven't thought about?

Reply:
The structure matches, my point was that the first and second parts (e.g. in the case of `metrics-apm-*`, I mean `metrics` and `apm`) are not intended to be changed. Changing them breaks things, because the privileges do not cater for that.

APM does not produce any data streams that match the patterns `metrics-apm-*` or `logs-apm-*`. There _is_ a data stream that matches `traces-apm-*`, hence why that has a different pattern in `kibana_system`.

Reply:
> @axw in this case, what would be your suggestion on naming aliases if following this documentation? https://www.elastic.co/guide/en/kibana/current/apm-spaces.html
>
> Changing it to something like metrics-apm.production-default would work with the current kibana_system privileges, but wouldn't it be more incidental given that typical APM data stream places other predefined things in place of production (e.g. metrics-apm.service_destination.1m-default).

My recommendation would be to put the environment in the namespace, so e.g. you may have `metrics-apm.all-production`. Or `metrics-apm.alias-production` as you suggested.

Sorry, I see where you're coming from now -- that additional ".all" or ".alias" feels a bit arbitrary. OTOH it's probably a good idea to name the alias something that won't ever be likely to collide with with a data stream name.

Anyway, per Dario above it sounds like the answer is that we have a bug in the UI and if that were fixed it would make this change unnecessary. Then you should be able to call the alias whatever you like, assuming your user role has necessary privileges.